### PR TITLE
Beehive rewrite (Floating point precision problems gone!)

### PR DESCRIPTION
**Pull request by hgentry**
## Description

I completely rewrote the math portion to use the spherical law of cosines to calculate angles and distances and the Haversine equation to calculate position. Everything is now calculated in reference to the center rather than being an iterative process, preventing float issues from stacking. Because of these changes, the result is very accurate, even at the poles.

## Motivation and Context

The floating point precision issues near the poles rendered it nearly unusable at certain latitudes.

## How Has This Been Tested?

I've been plotting worker points with a point plotting tool to check their locations. I've also run -st 2 -lp 2 locally, which works, but my machine can't handle more than that.

## Screenshots

-st 100 -lp 4 
map

-st 100 -lp 10
map

I also tried to render -st 100 -lp 100, but my machine couldn't handle 29700 location entries being drawn at once.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by hgentry** [comment]
Oh, come on, Travis! The "failure" was their fault, before it even entered the project.

File: Tools/Hex-Beehive-Generator/location_generator.py [hunk 1]
```diff
@@ -17,21 +17,20 @@
 wst = (int)(args.leaps)
 
 w_worker = (2 * st - 1) * r_hex
-d = 2 * w_worker
-d = math.sqrt(d**2+86.6**2)
+d = 2.0 * w_worker /1000.0
+d_s = d
+brng_s = 0.0
+brng = 0.0
+mod = math.degrees(math.atan(1.732/(6*(st-1)+3)))
```
Comment (reyqn) [inline Tools/Hex-Beehive-Generator/location_generator.py]: If we want to be accurate maybe we should stop approximations? 1.732 is math.sqrt(3)

**Comment by reyqn** [comment]
Also since you did this maybe you could look at the main search loop? It's still very inaccurate for high step numbers and it should be resolved with this method I think?
I did write #2375 quickly but it's pretty bad too.

**Comment by marauder37** [comment]
+1 from SameSexLove, let's go

**Comment by marauder37** [comment]
Actually this doesn't work, see: https://snag.gy/lO1Y8L.jpg

python location_generator.py -lat -33.9075685685 -lon 151.181878212 -st 5 -lp 3
-33.9075685685, 151.181878212
-33.8954226621, 151.18281253
-33.902169678, 151.194960059
-33.914315663, 151.194027391
-33.9197144438, 151.180943629
-33.9129660648, 151.168794715
-33.9008202685, 151.169730947
-33.8832767247, 151.183746582
-33.8711307561, 151.184680371
-33.8589847566, 151.185613894
-33.846838726, 151.186547153
-33.8346926643, 151.187480149
-33.8225465717, 151.18841288
-33.810400448, 151.189345348
-33.7982542933, 151.190277552
-33.7861081076, 151.191209493
-33.773961891, 151.19214117
-33.7618156433, 151.193072585
-33.7496693647, 151.194003737

I'll see if I can work out what's going on

**Comment by marauder37** [comment]
I suspect your mixed tabs/spaces indenting is messing with the turning code

**Comment by marauder37** [comment]
https://github.com/AHAAAAAAA/PokemonGo-Map/pull/2574

**Comment by Hoekynl** [comment]
So im running -st 6, what would the -lp value be, and what does the -lp value do? It isnt in the command line arguments

**Comment by billbo99** [comment]
I am running from the latest dev branch, and just doing 2x leeps I am seeing holes in the scanning locations.

image

**Pull request merged**